board.cpp

```diff
@@ -0,0 +1,205 @@
+#include "board.h"
+const int undefinedPerspective = -1;

```

Comment:
#should make `static`

#should use a different naming convention for constants; i.e. snake_case, UPPER_SNAKE_CASE, or kHungarianDromedaryCase. In this example, any of `undefined_perspective`, `UNDEFINED_PERSPECTIVE` or `kUndefinedPerspective`.

I suggest not to use snake_case, as that's commonly used with variables.